packages/calcite-components/src/components/list/list.tsx

```diff
@@ -328,12 +356,40 @@ export class List implements InteractiveComponent, LoadableComponent, SortableCo
     setComponentLoaded(this);
   }
 
+  disconnectedCallback(): void {
+    if (dragActive(this)) {
+      return;
+    }
+
+    this.disconnectObserver();
+    disconnectSortableComponent(this);
+    disconnectInteractive(this);
+    disconnectMessages(this);
+  }
+
   // --------------------------------------------------------------------------
   //
   //  Private Properties
   //
   // --------------------------------------------------------------------------
 
+  @State() effectiveLocale = "";
+
+  @Watch("effectiveLocale")
+  effectiveLocaleChange(): void {
+    updateMessages(this, this.effectiveLocale);
+  }
+
+  @Watch("effectiveLocale")
+  effectiveLocaleWatcher(): void {
```

Comment:
I think we can drop this watcher since the formatter is updated right before its used in `renderItemAriaLive`.